### PR TITLE
doc: nrf5340audio: remove HW info from app doc

### PR DESCRIPTION
**Pull request by greg-fer**
Removed hardware-related sections from the application readme. These were moved to Infocenter and Zephyr upstream. OCT-2395.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

----

- [x] Wait for publication in Infocenter; update this PR
- [x] Update changelog with an entry

### CODE REVIEW

**Review by koffes — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -160,7 +160,8 @@ nRF5340 Audio
 
   * Power module has been re-factored so that it uses upstream Zephyr INA23X sensor driver.
   * BIS headsets can now switch between two broadcast sources (two hardcoded broadcast names).
-  * Documentation in the :ref:`nrf53_audio_app_ui` and :ref:`nrf53_audio_app_testing_steps_cis` sections with information about using **VOL** buttons to switch headset channels.
+  * :ref:`nrf53_audio_app_ui` and :ref:`nrf53_audio_app_testing_steps_cis` sections in the application documentation with information about using **VOL** buttons to switch headset channels.
```
Comment (erikrobstad) [inline doc/nrf/releases/release-notes-changelog.rst:163]: Is a rebase required?

**Review by erikrobstad — COMMENTED** [review]
"Hardware requirements for external flash memory DFU" is in the Infocenter:
https://infocenter.nordicsemi.com/topic/ug_nrf5340_audio/UG/nrf5340_audio/hw_external_memory.html

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/releases/release-notes-changelog.rst:163]: No, I've edited this when fixing the rebase.

File: applications/nrf5340_audio/README.rst [hunk 2]
```diff
@@ -309,299 +309,13 @@ Software codec requirements
...
-    * - Rev 1.0.0
-      - CS47L63 AD-DA converter (**U2**) may fail to start
-      - In some occasions, the 1.2 V power supply for **U2** is not provided at boot-up.
-        This is caused by higher than expected inrush current.
-        This function is tested in production.
-        The issue should not happen, although we observe that some kits have the problem.
-      - Restart kit or attach the battery to the kit before connecting the USB cable.
-        If problem persists, contact Nordic Semiconductor and ask for replacement.
-      - Rev 1.0.1
+Read the `nRF5340 Audio DK Hardware`_ documentation on Nordic Semiconductor Infocenter for more information about this development kit.
```
Comment (erikrobstad) [inline applications/nrf5340_audio/README.rst:318]: Maybe?:

```suggestion
Read the `nRF5340 Audio DK`_ hardware documentation on Nordic Semiconductor Infocenter for more information about this development kit.
```

**Review by erikrobstad — APPROVED** [review]

File: applications/nrf5340_audio/README.rst (hunk 2 above)
Comment (greg-fer) [reply applications/nrf5340_audio/README.rst:318]: Nope, that's the title of the section on Infocenter... :(

**Pull request merged**